cli/Sources/TuistServer/Services/PutCacheValueService.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+import Mockable
+import OpenAPIRuntime
+import OpenAPIURLSession
+
+@Mockable
+public protocol PutCacheValueServicing: Sendable {
+    func putCacheValue(
+        casId: String,
+        entries: [String: String],
+        fullHandle: String,
+        serverURL: URL
+    ) async throws
+}
+
+public enum PutCacheValueServiceError: LocalizedError {
+    case unknownError(Int)
+    case unauthorized(String)
+    case forbidden(String)
+    case notFound(String)
+    case putValueFailed
+
+    public var errorDescription: String? {
+        switch self {
+        case let .unknownError(statusCode):
+            return "The CAS value could not be stored due to an unknown Tuist response of \(statusCode)."
+        case let .unauthorized(message), let .forbidden(message), let .notFound(message):
+            return message
+        case .putValueFailed:
+            return "The CAS value storage failed due to an unknown error."
+        }
+    }
+}
+
+public final class PutCacheValueService: PutCacheValueServicing {
+    private let fullHandleService: FullHandleServicing
+
+    public convenience init() {
+        self.init(
+            fullHandleService: FullHandleService()
+        )
+    }
+
+    init(
+        fullHandleService: FullHandleServicing
+    ) {
+        self.fullHandleService = fullHandleService
+    }
+
+    public func putCacheValue(
+        casId: String,
+        entries: [String: String],
+        fullHandle: String,
+        serverURL: URL
+    ) async throws {
+        let client = Client.authenticated(serverURL: serverURL)
+        let handles = try fullHandleService.parse(fullHandle)
+
+        let entriesPayload = Operations.putCacheValue.Input.Body.jsonPayload.entriesPayload(
+            entries
+                .filter { $0.key == "value" }
```

Comment:
there might be other keys than value ... but I haven't seen any. My guess is that they made this a dictionary instead of specific fields, so this contract is more future-proof. We can update this if we see other keys than value in the future, but for now, it made things easier to work with if we assume the keys are all `value` instead of having to handle a generic dictionary.